demos/projects/ESPRESSIF/aziotkit/main/azure_iot_freertos_esp32_sensors.c

```diff
@@ -0,0 +1,390 @@
+#define sampleazureiotgsgPROCESSOR_ARCHITECTURE_PROPERTY_NAME    ( "processorArchitecture" )
+#define sampleazureiotgsgPROCESSOR_MANUFACTURER_PROPERTY_NAME    ( "processorManufacturer" )
+#define sampleazureiotgsgTOTAL_STORAGE_PROPERTY_NAME             ( "totalStorage" )
+#define sampleazureiotgsgTOTAL_MEMORY_PROPERTY_NAME              ( "totalMemory" )
+
+const char * pcManufacturerPropertyValue = "ESPRESSIF";
+const char * pcModelPropertyValue = "ESP32 Azure IoT Kit";
+const char * pcSoftwareVersionPropertyValue = "1.0.0";
+const char * pcOsNamePropertyValue = "FreeRTOS";
+const char * pcProcessorArchitecturePropertyValue = "ESP32 WROVER-B";
+const char * pcProcessorManufacturerPropertyValue = "ESPRESSIF";
+const double xTotalStoragePropertyValue = 4153344;
+const double xTotalMemoryPropertyValue = 8306688;
+
+/**
+ * @brief Telemetry Values
+ */
+#define sampleazureiotTELEMETRY_TEMPERATURE        ( "temperature" )
+#define sampleazureiotTELEMETRY_HUMIDITY           ( "humidity" )
+#define sampleazureiotTELEMETRY_LIGHT              ( "light" )
+#define sampleazureiotTELEMETRY_PRESSURE           ( "pressure" )
+#define sampleazureiotTELEMETRY_ALTITUDE           ( "altitude" )
+#define sampleazureiotTELEMETRY_MAGNETOMETERX      ( "magnetometerX" )
+#define sampleazureiotTELEMETRY_MAGNETOMETERY      ( "magnetometerY" )
+#define sampleazureiotTELEMETRY_MAGNETOMETERZ      ( "magnetometerZ" )
+#define sampleazureiotTELEMETRY_PITCH              ( "pitch" )
+#define sampleazureiotTELEMETRY_ROLL               ( "roll" )
+#define sampleazureiotTELEMETRY_ACCELEROMETERX     ( "accelerometerX" )
+#define sampleazureiotTELEMETRY_ACCELEROMETERY     ( "accelerometerY" )
+#define sampleazureiotTELEMETRY_ACCELEROMETERZ     ( "accelerometerZ" )
+
+static time_t xLastTelemetrySendTime = INDEFINITE_TIME;
+
+/**
+ * @brief Command Values
+ */
+#define sampleazureiotCOMMAND_EMPTY_PAYLOAD        "{}"
+
+static const char sampleazureiotCOMMAND_TOGGLE_LED1[] = "ToggleLed1";
+static const char sampleazureiotCOMMAND_TOGGLE_LED2[] = "ToggleLed2";
+static const char sampleazureiotCOMMAND_DISPLAY_TEXT[] = "DisplayText";
+
+static bool xLed1State = false;
+static bool xLed2State = false;
+
+/**
+ * @brief Property Values
+ */
+#define sampleazureiotPROPERTY_STATUS_SUCCESS      200
+#define sampleazureiotPROPERTY_SUCCESS             "success"
+#define sampleazureiotPROPERTY_TELEMETRY_FREQUENCY ( "telemetryFrequencySecs" )
+
+static int lTelemetryFrequencySecs = 2;
+
+static uint8_t ucPropertyPayloadBuffer[ 384 ];
+/*-----------------------------------------------------------*/
+
+int32_t lGenerateDeviceInfo( uint8_t * pucPropertiesData,
+                             uint32_t ulPropertiesDataSize )
+{
+    AzureIoTResult_t xAzIoTResult;
+    AzureIoTJSONWriter_t xWriter;
+    int32_t lBytesWritten;
+
+    /* Update reported property */
+    xAzIoTResult = AzureIoTJSONWriter_Init( &xWriter, ucPropertyPayloadBuffer, sizeof( ucPropertyPayloadBuffer ) );

```

Comment:
this is using a private buffer, so the payload is never returned from the function.